### ISSUE TITLE
0.5.0 - any app button turns off the TV

### ISSUE THREAD

**Issue by Stooovie** [usernein/tv-card#10]
**Release with the issue:**
0.5.0

**Last working release (if known):**
0.4.2

**Browser and Operating System:**
Any browser or HA app

**Description of problem:**
Tapping any app button in this addon causes the TV to turn off

TV should launch the respective app

**Javascript errors shown in the web inspector (if applicable):**

```

```

**Additional information:**
Fine after downgrading to 0.4.2. My TV is a 2018 Samsung NU8000

**Comment by usernein** [comment]
Any button? Even the native ones?
Or only with custom buttons for apps?

**Comment by usernein** [comment]
Can you share the config you are using, so i can try to replicate it here?
It happened to me only once, randomly

**Comment by tretabyte** [comment]
Hi,

it's happening with any app button, custom or default (netflix and youtube also shuts down the TV).
config you mean the card?
this is what I have atm:
```type: custom:tv-card
entity: media_player.samsungqn900a65
title: Samsung QN900A65 (Sala)
power_row:
  - power
channel_row:
  - channel_up
  - info
  - channel_down
apps_row:
  - netflix
  - youtube
  - hbomax
  - plex
custom_sources:
  hbomax:
    icon: hbo
    source: HBO Max
  plex:
    icon: plex
    source: Plex
custom_icons:
  hbo: >-
    M7.042 16.896H4.414v-3.754H2.708v3.754H.01L0
    7.22h2.708v3.6h1.706v-3.6h2.628zm12.043.046C21.795 16.94 24 14.689 24
    11.978a4.89 4.89 0 0 0-4.915-4.92c-2.707-.002-4.09 1.991-4.432
    2.795.003-1.207-1.187-2.632-2.58-2.634H7.59v9.674l4.181.001c1.686 0
    2.886-1.46 2.888-2.713.385.788 1.72 2.762 4.427 2.76zm-7.665-3.936c.387 0
    .692.382.692.817 0 .435-.305.817-.692.817h-1.33v-1.634zm.005-3.633c.387 0
    .692.382.692.817 0 .436-.305.818-.692.818h-1.33V9.373zm1.77
    2.607c.305-.039.813-.387.992-.61-.063.276-.068 1.074.006
    1.35-.204-.314-.688-.701-.998-.74zm3.43 0a2.462 2.462 0 1 1 4.924 0 2.462
    2.462 0 0 1-4.925 0zm2.462 1.936a1.936 1.936 0 1 0 0-3.872 1.936 1.936 0 0 0
    0 3.872Z
  plex: M11.643 0H4.68l7.679 12L4.68 24h6.963l7.677-12-7.677-12
volume_row: buttons
navigation_row: buttons
source_row:
  - return
  - home
  - source
media_control_row:
  - rewind
  - play
  - pause
  - fast_forward
```

**Comment by usernein** [comment]
I think i got it.
How do you usually turn on the TV? Is it using tv-card?

I managed to reproduce the issue here by turning on the TV using the physical button then immediately trying to use an app button. The tv turned off and these lines were logged:
<img width="392" alt="Captura de Tela 2022-07-12 às 19 21 56" src="https://user-images.githubusercontent.com/29507335/178607990-aead7c49-9fba-420e-b1f6-033370287f8b.png">

My theory is that it turns off when we immediately try to change the source before the connection with the tv is stablished (for me it takes about 30 seconds)

If i'm right, it has nothing to do with tv-card, but with this the samsung integration we use.
But also if i'm right, i don't know why it started to happen at v0.5.0 (is it possible that it always happened and we didn't notice before and coincidentally made the tests pass at v0.4.2 all this time?)

I'll keep testing and trying to figure it out

**Comment by usernein** [comment]
The big mistery is because i didn't touch even near the core of changing sources in v0.5.0

**Comment by tretabyte** [comment]
what I noticed yesterday was that while the TV was completely off, hitting the On button on the card wasn't doing much but as soon I hit the Netlflix button it imedially turned the TV on, hitting it again turned it Off
is there anything extra I could do from my side to help you out on this?

**Comment by usernein** [comment]
The delay on turning on is because the samsung integration takes some time to establish the connection and to ensure it's really turned on

Using tv-card's default power key, you may click it like 3 times with some interval, until you get it
That's why i gave up with it (i can't do much to help on tv-card) and started using the physical button or a tuya smart ir i had at home

Maybe the Netflix button did not turn on, but the power button did, with delay
Also, maybe Netflix actually did, it's still a mistery

> is there anything extra I could do from my side to help you out on this?
Can you test these situations below? I'm also doing it:
1. Using v0.4.2, with the tv completely off (i mean, passed about 2 minutes after turned it off, to ensure the connection is closed), turn it on through your remote control or physical button (any non-smart way) and immediately hit any app button after the tv is turned on. Check if it turns off
2. Using v0.5.0, with the tv completely on (passed 1 minute after turned it on, to ensure the connection is established), try to use any app button. If possible, check the state of the tv entity before using tv-card, to make sure it's on and connected. Check if it turns off.

If it turns off on any situation, please check the logs at Settings > System > Logs and see what happened

**Comment by usernein** [comment]
> hitting the On button on the card wasn't doing much but as soon I hit the Netflix button it immediately turned the TV on

@tretabyte you are right, the apps buttons instantly turn the tv on
I turned it on using YouTube, but this time it didn't turn off the tv after clicking it again

I'm getting more sure it's on the integration side

**Comment by usernein** [comment]
@Stooovie are you using samsung too? Which integration?

**Comment by tretabyte** [comment]
weird.. rollback to 0.4.2 and tested, app buttons doens't work, updated to 0.5.2 (was using 0.5.0), app buttons still don't work
the only thing spotted on logs is this:
```This error originated from a custom integration.

Logger: custom_components.samsungtv_smart.media_player
Source: custom_components/samsungtv_smart/media_player.py:1537
Integration: SamsungTV Smart (documentation, issues)
First occurred: 01:48:03 (6 occurrences)
Last logged: 01:49:24

Unsupported source```

**Comment by Stooovie** [comment]
> @Stooovie are you using samsung too? Which integration?

Yes I am, 2018 NU8000 via [this integration](https://github.com/ollo69/ha-samsungtv-smart)
